src/unix/mmap.c

```diff
@@ -169,6 +168,33 @@ sysreturn mremap(void *old_address, u64 old_size, u64 new_size, int flags, void
     if (new_size <= old_size)
         return sysreturn_from_pointer(old_address);
 
+    /* verify we have a single vmap for the old address range */
+    vmap old_vm = (vmap)rangemap_lookup(p->vmaps, old_addr);
+    if ((old_vm == INVALID_ADDRESS) ||
+        (!range_contains(
+            old_vm->node.r,
+            irange(old_addr, old_addr + old_size)
+        )))
+        return -EFAULT;
+
+    /* XXX should determine if we're extending a virtual32 allocation...
+     * - for now only let the user move anon mmaps 
+     */
+    if (! ((old_vm->flags & VMAP_FLAG_MMAP) &&
+           (old_vm->flags & VMAP_FLAG_ANONYMOUS))
+       )
+    {
+        msg_err("mremap only supports anon mmap regions at the moment\n"); 
```

Comment:
Would this not work for mappings backed by a file? At least at this time all the pages should be present for a backed mapping.

Reply:
Yeah I don't see why not.  As soon as we see something that needs it, should just be a one line fix